server/src/main/java/io/kroki/server/service/D2.java

```diff
@@ -89,21 +89,26 @@ public void convert(String sourceDecoded, String serviceName, FileFormat fileFor
   private byte[] d2(byte[] source, JsonObject options) throws IOException, InterruptedException, IllegalStateException {
     List<String> commands = new ArrayList<>();
     commands.add(binPath);
-    String value = options.getString("theme");
-    if (value != null) {
+    String theme = options.getString("theme");
+    if (theme != null) {
       int themeId = 0;
-      Integer builtinThemeId = builtinThemes.get(value.toLowerCase().replaceAll("\\s", "-"));
+      Integer builtinThemeId = builtinThemes.get(theme.toLowerCase().replaceAll("\\s", "-"));
       if (builtinThemeId != null) {
         themeId = builtinThemeId;
       } else {
         try {
-          themeId = Integer.parseInt(value, 10);
+          themeId = Integer.parseInt(theme, 10);
         } catch (NumberFormatException e) {
           // ignore, fallback to 0
         }
       }
       commands.add("--theme=" + themeId);
     }
+    String layout = options.getString("layout");
+    if (layout != null && layout.equals("elk")) {
```

Comment:
(nitpick/suggestion): can be simplified to:

```suggestion
    if ("elk".equals(layout)) {
```

or:

```suggestion
    if (Objects.equals(layout, "elk")) {
```

But I'm fine with your code.

Reply:
I initially wrote it like your first suggestion, but I was getting a null pointer error when `layout` wasn't defined. I'm not actually that familiar with Java but I assume that this is because the `equals()` function doesn't check for null pointers before dereferencing. It only started working properly once I added the explicit check for null values first. 🤷🏻‍♂️

Reply:
> I initially wrote it like your first suggestion, but I was getting a NullPointerException when layout wasn't defined.

You need to call `equals` on a non-null object:


```java
if ("elk".equals(layout)) { } // OK
```

```java
if (layout.equals("elk")) { } // NullPointerException if layout is null
```

Arguably, the first option does not read well since we want to know if layout is equals to "elk" (the opposite sounds odd "if elk is equals to layout").

Anyway, I'm fine with it 😉